litmus-portal/graphql-server/pkg/chaos-workflow/ops/operations.go

```diff
@@ -329,3 +371,41 @@ func SendWorkflowEvent(wfRun model.WorkflowRun, r *store.StateData) {
 	}
 	r.Mutex.Unlock()
 }
+
+// ResiliencyScoreCalculator calculates the Rscore and returns the execdata string
+func ResiliencyScoreCalculator(execData string, wfid string) string {
+	var resiliency_score, weightSum, totalTestResult, totalExperiments, totalExperimentsPassed int = 0, 0, 0, 0, 0
+	var jsonData WorkflowEvent
+	json.Unmarshal([]byte(execData), &jsonData)
+	for _, value := range jsonData.Nodes {
+		if value.Type == "ChaosEngine" {
+			if value.ChaosExp == nil {
+				continue
+			}
+			chaosWorkflows, _ := dbOperationsWorkflow.GetWorkflows(bson.D{{"workflow_id", bson.M{"$in": []string{wfid}}}})
```

Comment:
Move this outside loop as wfid will remain same, currently the db is queried multiple times for the same data.

Reply:
wfid is unique so only 1 record will be returned. Directly use `chaosWorkflows[0]`